web/src/LogStore.ts

```diff
@@ -59,7 +59,17 @@ class StoredLine {
   }
 }
 
-type callback = () => void
+export enum LogUpdateAction {
+  append,
+  truncate,
+}
+
+export interface LogUpdateEvent {
+  action: LogUpdateAction
+  spans: { [key: string]: Proto.webviewLogSpan }
```

Comment:
what problem does this field solve? this seems like it's adding extra bookkeeping but not providing any real benefit.

Reply:
It was for in conjunction with conditionally triggering an update by the receiver but is unnecessary in the end (see other comment) - got rid of it and associated bookkeeping.